protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -226,30 +322,52 @@ fn execute_cheatcode_hint(
             panic_data_end,
         } => {
             let contract_address = get_val(vm, prepared_contract_address)?;
+            // TODO deploy should fail if contract address provided doesn't match calculated
+            //  or not accept this address as argument at all.
```

Comment:
Is there an issue for this?

Reply:
Updated #1991 to include it in title and description. Will link it in this TODO.